### DISCUSSION TITLE
How to use SCENIC when comparing dataset representing two conditions?

### DISCUSSION THREAD

**Discussion by singcell** [aertslab/SCENIC#261] [* General SCENIC questions]
I have control and treated samples and I have generated Differential gene expression list from  Seurat integration tool. I wanted to determine the gene regulatory network associated with the genes that are differentially expressed between conditions.
How can I use SCENIC tool when I have two different dataset comparing two different conditions. Do I need to add samples separately ?
Do you have workflow for this kind of analysis.

**Comment by saeedfc** [comment]
Hi,
I am not a contributor to this package. But If I may suggest, I will start with feeding the Normalized matrix of the whole data(integrated data) (RNA assay, slot = "data") to SCENIC.

Onc you generate the results, get the AUC matrix. In this matrix you have the 'activity scores' for each transcription factor for each cell (Transcription factors as columns and cells as rows). You can add this to the meta.data of your seurat object. Then you can visualize using ViolinPlots with split.by argument for each transcription factor to see the difference between Treated and Control. Once you add the AUC output to your seurat meta.data, there are a lot of possibilities.

If I understand it correctly, to make a comparison possible, you should feed the  whole matrix together and not separate and run SCENIC once.

You may also look at the regulon specificity score functionality available with PySCENIC. After adding the AUC output in the meta.data, you may split the AUC matrix (treated vs control), compute regulon specificity score for each AUC matrix separately and try to see the differences.

to compute regulon specificity score you need AUC matrix (treated  and control) the cluster information of your seurat clusters. So you need meta data also split between treated and control. Since the AUC essentially xome from the single analysis, this can be a good comparison

Hope it helps

**Reply by lucygarner** [reply]
The developers generally recommend to use raw counts rather than normalised counts.

**Reply by MariaRosariaNucera** [reply]
Hi, thanks for your comment, could you elaborate more about the reasoning behind using the whole matrix and not separate?

**Comment by singcell** [comment]
@saeedfc Thank you for your help.

**Comment by PDhingra2016** [comment]
Hi @singcell
Were you able to compare the two datasets? I have to compare a cell type between disease and control. I would like to see if there are differences in regulons in disease vs control. As I am new to SCENIC, was looking for some scripts to incorporate this with my seurat pipeline.
Looking forward to your help
Thanks

**Reply by RuiqingZ1** [reply]
Hi PDhingra2016,
I’m running into the same issue. I aim to compare differences in eRegulons between treatment conditions for each cell type. I’m wondering if you’ve found a solution to this problem.
Thank you!

**Comment by saeedfc** [comment]
Just passed by this again !
It is best to add the AUC matrix as a new assay to your seurat object. This give way more possibilities.
`SeuObj[["scenic"]] <- CreateAssayObject(data =  t(auc_mtx)); 
`remember to not normalize this assay. Transposing only because, by default, this assay comes with Transcription factors as column names in the pyscenic output.
Saeed

**Reply by JoyOtten** [reply]
I'm wondering where you actually can find the auc_mtx if you're working with R. I'm trying to locate the dataframe but honestly it's not an easy find.

**Reply by saeedfc** [reply]
There is a function called get_AUC in R

**Reply by JoyOtten** [reply]
I got it actually the function is called getAUC in R but many thanks

**Comment by x1han** [comment]
Hello everyone, I have a question regarding this issue. For instances such as gene editing, shouldn't the target genes be different for the same TF in the case group and the control group? (As far as I understand, the target genes are calculated.) So, in this situation, when comparing two conditions, wouldn't it be better to analyze each group separately? However, if we analyze the two groups separately, how should we proceed with the comparison?

**Comment by lzh-fugwaa** [comment]
Hello everyone, I have a samll question regarding this issue. I have some group from different Time Point, and I want to compare the differences in predicted transcription factors between these group. Meanwhile, I have different cell types, and I need to do cell type by cell type analysis when I compare different Time Point? Or all the cell types together?

**Reply by RuiqingZ1** [reply]
Hi Fugwaaaa, 
I’m running into the same issue. I aim to compare differences in eRegulons between treatment conditions for each cell type. I’m wondering if you’ve found a solution to this problem.
Thank you!

**Reply by AnnaAntonatouPap** [reply]
Hey @RuiqingZ1 ,

I am aiming to do exactly the same. Have you figured it out ?

Thanks in advance,
Anna
